completions.go

```diff
@@ -1,34 +1,88 @@
 package kingpin
 
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type completion struct {
+	flags []string
+	words []string
+}
+
+const (
```

Comment:
I don't quite understand the purpose of these. Are they placeholders somehow? This could use some explanatory comments.